packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
+
+export interface CoreDashboardModel {
+  panels: CorePanelModel[];
+  uid: string;
+  title: string;
+  updatePanels: (panels: CorePanelModel[]) => unknown;
+}
+
+export interface CoreDashboardSrv {
+  dashboard?: CoreDashboardModel;
+}
+
+// Public types
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ *
+ * @public
+ */
+export interface PluginsAPIDashboardSrv {
+  /**
+   * @deprecated use `getCurrentDashboard()` instead
+   */
+  dashboard?: PluginsAPIDashboardModel;
+  /**
+   * Represents the currently loaded dashboard. Returns 'undefined' if no dashboard is loaded.
+   */
+  getCurrentDashboard(): PluginsAPIDashboardModel | undefined;
+}
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ * @public
+ */
+export interface PluginsAPIDashboardModel {
+  /**
+   * The uid of the dashboard
+   * @remarks
+   * Note: The old `id` attribute is deprecated and therefore not included in the public API.
+   * @readonly
+   */
+  uid: string;
+  /**
+   * The title of the dashboard
+   * @readonly
+   */
+  title: string;
+  /**
+   * @readonly
+   * @deprecated use `getPanels()`
+   */
+  panels: PluginsAPIPanelModel[];
+
+  /**
+   * Returns the panels within the current dashboard.
+   *
+   * Note: Modifications to the panels are not immediately visible on the dashboard and are not automatically saved.
+   *
+   * Users must manually save the dashboard to persist any changes.
+   */
+  getPanels(): PluginsAPIPanelModel[];
```

Comment:
Could this function be exposed directly from the `dashboardSrv` instead of nesting them in a model?

Reply:
Panels belong to a dashboard. I think it makes sense that you get the dashboard from the dashboard service and you get the panels from the dashboard itself. 

That's also the same way we do it in grafana-core, `dashboardSrv` provides dashboards and you get the panels from the dashboard object.

I rather keep it consistent and following the logic of hierarchy in this case.

Reply:
Resolution: We are going for a flat API where all the methods are part of `dashboardSrv`